skyportal/handlers/api/mpc.py

```diff
@@ -0,0 +1,197 @@
+import arrow
+from astropy.time import Time
+from astropy.coordinates import Angle
+import astropy.units as u
+import asyncio
+from bs4 import BeautifulSoup
+import requests
+import re
+from sqlalchemy.orm import sessionmaker, scoped_session
+from tornado.ioloop import IOLoop
+import urllib
+
+from baselayer.app.env import load_env
+from baselayer.app.access import auth_or_token
+from baselayer.log import make_log
+
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    Obj,
+    User,
+)
+
+env, cfg = load_env()
+log = make_log('api/mpc')
+
+Session = scoped_session(sessionmaker())
+
+MPC_ENDPOINT = cfg['app.mpc_endpoint']
+mpcheck_url = urllib.parse.urljoin(MPC_ENDPOINT, 'cgi-bin/mpcheck.cgi')
+
+
+class ObjMPCHandler(BaseHandler):
+    @auth_or_token
+    def post(self, obj_id):
+        """
+        ---
+        description: Retrieve an object's status from Minor Planet Center
+        tags:
+          - objs
+        parameters:
+          - in: path
+            name: obj_id
+            required: true
+            schema:
+              type: string
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  obscode:
+                    type: string
+                    description: |
+                      Minor planet center observatory code.
+                      Defaults to 500, corresponds to geocentric.
+                  date:
+                    type: string
+                    description: |
+                      Time to check MPC for.
+                      Defaults to current time.
+                  limiting_magnitude:
+                    type: float
+                    description: |
+                      Limiting magnitude down which to search.
+                      Defaults to 24.0.
+                  search_radius:
+                    type: float
+                    description: |
+                      Search radius for MPC [in arcmin].
+                      Defaults to 1 arcminute.
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        date = data.get('date')
+        if date is None:
+            date = Time.now()
+        else:
+            try:
+                date = Time(arrow.get(date).datetime, format='datetime')
+            except (TypeError, arrow.ParserError):
+                return self.error(f'Cannot parse time input value "{date}".')
+
+        limiting_magnitude = data.get('limiting_magnitude', 24.0)
+        try:
+            limiting_magnitude = float(limiting_magnitude)
+        except Exception:
+            return self.error('Cannot read in limiting magnitude.')
+
+        search_radius = data.get('search_radius', 60.0)
```

Comment:
The default is one minute but if the user inputs 1 or 2 that would not be in minutes...

Reply:
good catch